src/vse_sync_pp/analyze.py

```diff
@@ -0,0 +1,68 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+import sys
+import json
+import pandas as pd
+from argparse import ArgumentParser
+
+from .common import open_input
+
+from .parse import PARSERS
+from .analyzers import ppsdpll
+from . plot import Plotter
+
+ANALYZERS = {
+    cls.id_: cls for cls in (
+        ppsdpll.PhaseOffsetTimeError,
+        ppsdpll.ConstantPhaseOffsetTimeError,
+        ppsdpll.PhaseOffsetTimeDeviation,
+        ppsdpll.MaxPhaseOffsetTimeIntervalError
+    )
+}
+def analyzer2parser(analyzer_id):
+    """Map the analyzer to the specific parser"""
+    if "dpll" in analyzer_id:
+     return "dpll/phase-offset"
+    else:
+     return "ts2phc/time-offset"
+
+def main():
+    """Print Synchronization KPIs from parsed data to stdout"""
+    aparser = ArgumentParser(description=main.__doc__)
+    aparser.add_argument(
+        '-c', '--canonical', action='store_true',
+        help="parse canonical log data from input",
+    )
+    aparser.add_argument(
+        'input',
+        help="data parsed or '-' to read from stdin",
+    )
+    aparser.add_argument(
+        '-p','--parser', choices=tuple(PARSERS),
+        help="data to parse from the log file",
+    )
+    aparser.add_argument(
+        'analyzer', choices=tuple(ANALYZERS),
+        help="analyzers to run",
+    )
+    aparser.add_argument(
+        'output',
+        help="filename to output image to",
+    )
+    args = aparser.parse_args()
+    if args.parser is not None:
```

Comment:
Is this option currently needed? The function mapping parser from analyzer is complete: when would we need to override that? Feels like it can be safely deleted.